.github/workflows/dotnet-tests.yml

```
@@ -0,0 +1,29 @@
+name: .NET Tests
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+
+jobs:
+  build-and-test:
+    runs-on: windows-latest
```

Я бы предложил запускать не на Windows так как такие тесты будут очень долго запускаться.
```suggestion
    runs-on: ubuntu-latest
```